### PR TITLE
Create the pets table, add a seed file to add initial data to the pets table

### PR DESCRIPTION
**Pull request by dvwhite**
I added a `knex` migration to create the `pets` table, which will be used to store data about the client's pets. Please see the table schema in the image below:

Table schema

I created a seed file to insert initial data into the pets table. This is intended to assist developers by providing readily available data for use in testing either the React app or the REST API.

I also added node controllers for full CRUD, and middleware to authorize changes to pet data by verifying pet ownership before updating or deleting a pet.

I added database models providing full CRUD operations against the `/pets` table.

### CODE REVIEW

File: api/pet/petRouter.js [hunk 1]
```diff
@@ -0,0 +1,396 @@
...
+const petsDb = require('../pet/petModel');
+
+// Utilities
+const { errDetail } = require('../utils/utils');
+
+// Auth middleware
+const authRequired = require('../middleware/authRequired');
+
+/**
+ * @swagger
```
Comment (hmerritt) [inline api/pet/petRouter.js:14]: Props for writing documentation for each route.

I've never used swagger before and I'm curious to see what it looks like when compiled.

File: data/seeds/002_locations.js [hunk 2]
```diff
@@ -3,23 +3,23 @@ const faker = require('faker');
...
-    groomerId: 'bftqzfqkq3xzj953q64r',
+    groomerId: '00ultwz1n9ORpNFc04x6',
     businessName: "Gillian's Fine Pet Grooming",
     address: faker.address.streetAddress('###'),
     email: faker.internet.email(),
     phoneNumber: faker.phone.phoneNumberFormat(),
-    lat: faker.address.latitude(),
-    lng: faker.address.longitude(),
+    lat: -77.0497,
+    lng: 38.9007,
```
Comment (hmerritt) [inline data/seeds/002_locations.js:22]: Thanks for adding the real lat-lng address

**Review by hmerritt — APPROVED** [review]
Great work getting the pets endpoint up and running, plus adding tests and documentation.

File: api/pet/petRouter.js [hunk 3]
```diff
@@ -0,0 +1,396 @@
...
+      petId,
+      pet,
+    };
+    next();
+  } catch (err) {
+    // Handle errors with a custom error utility
+    errDetail(err);
+  }
+}
+
```
Comment (Cberumen51) [inline api/pet/petRouter.js:395]: Nice job writing all the CRUD operations

File: data/seeds/001_profiles.js [hunk 4]
```diff
@@ -1,22 +1,65 @@
...
-}));
-
 exports.seed = function (knex) {
-  // Deletes ALL existing entries
   return knex('profiles')
     .del()
     .then(function () {
-      // Inserts seed entries
-      return knex('profiles').insert(profiles);
+      return knex('profiles').insert([
```
Comment (Cberumen51) [inline data/seeds/001_profiles.js:7]: nice taking out all the tests that were given to usu and using our own

**Review by Cberumen51 — APPROVED** [review]
Nice job on everything that was changed and added. it was a pretty big pull request

**Pull request merged**